### DISCUSSION TITLE
✔ Online scanner with best practices for Security Headers

### DISCUSSION THREAD

**Discussion by bartosz-io** [oshp/oshp-tracking#18] [Ideas]
Hi!

My name is Bartosz and I created online training for web developers - websecurity-academy.com.
I bought the domain httpScanner.com and had been working on a free online scanner that will help developers evaluate the HTTP security profile of web applications.

The general idea is to put into a nice UI/UX **all the best practices and educate developers for free with guides/tutorials** regarding HTTP security headers.

The main screen is just a simple input text and a submit button that redirects to a report page.
Here is a screen of a report prototype:
<img width="1249" alt="scanner" src="https://user-images.githubusercontent.com/8076187/224493856-ca7ad386-10d5-4f46-91fa-eb7b6bd2a31a.png">

I wanted to ask you about the feedback - do you think this free resource is worth creating?

Thank you!

**Comment by bartosz-io** [comment]
Here is why I pinged you on LinkedIn @righettod :)

**Comment by righettod** [comment]
Hi @bartosz-io 

😉 Thanks a lot for opening this discussion thread.

🥇 Very nice project and initiative:  websecurity-academy + httpScanner .

❤ **For me, every resources that can helps DEV and OPS team to leverage the HTTP security headers is useful.**

💡 For httpScanner, a suggestion can be to differ from https://securityheaders.com/ by focusing on providing a better output/feedback that an DEV/OPS can directly leverage.

👀 Some examples:
* If you detect than some headers are missing AND you identify that the remote web server is an Nginx then you can provide the code snippet of the configuration to add the missing headers.
* Same idea regarding headers disclosing technical informations.

📡 To help you, we provided such references JSON file:

![image](https://user-images.githubusercontent.com/1573775/224532817-6d4bd1c3-c048-4554-a4ae-6c03043828c8.png)

✅ So for me, **YES** your projects are 100% useful.

🥰 Once ready, we will add it to the **Technical Resources** tab and notify the community about this new tool.

@riramar  What is your point of view ?

**Comment by bartosz-io** [comment]
@righettod Thank you for your feedback! This is awesome ♥️

We are starting the development next week!

**Comment by righettod** [comment]
You're welcome 😀

**Comment by bartosz-io** [comment]
Also, I really appreciate the suggestions like:

> focusing on providing a better output/feedback that an DEV/OPS can directly leverage.

If you have any additional points here - please share. Thank you! 🦾

**Comment by riramar** [comment]
@righettod I agree with you. I don't have much to add.

**Comment by righettod** [comment]
✅ Consider as completed due to past time without interactions.
